code/addons/test/src/node/vitest-manager.ts

```diff
@@ -276,6 +273,9 @@ export class VitestManager {
     this.updateLastChanged(id);
     this.storyCountForCurrentRun = 0;
 
+    if (!this.testManager.watchMode) {
+      return;
+    }
```

Comment:
The key was to return _here_, _after_ we've invalidated caches and whatnot above.

Initially I returned immediately in this function, but that created wrong coverage results too, so we have to return just before running tests.

Reply:
Maybe add a comment about the relevancy of returning here.